mcp2210-core.c

```diff
@@ -709,6 +712,358 @@ static inline int eeprom_read_complete(struct mcp2210_cmd *cmd_head,
 }
 #endif /* CONFIG_MCP2210_CREEK */
 
+#ifdef CONFIG_OF
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief This function is used to parse the device tree node for mcp2210
+ *        if available and based on that configure the mcp2210 bridge.
+ *
+ * @Param dev mcp2210 device struct pointer
+ *
+ * @Returns  0 on success, negative values on error
+ */
+/* ----------------------------------------------------------------------------*/
+static int mcp2210_parse_dts(struct mcp2210_device *dev)
+{
+    struct device_node *np;
+    struct mcp2210_board_config board_config;
+    struct mcp2210_chip_settings chip_settings;
+    struct mcp2210_chip_settings powerup_chip_settings;
+    struct mcp2210_spi_xfer_settings spi_settings;
+    struct mcp2210_spi_xfer_settings powerup_spi_settings;
+    struct mcp2210_usb_key_params usb_key_params;
```

Comment:
This is probably more data that we should be putting on the stack. Consider define a single struct and then make one allocation:

```
struct mcp2210_dts_data { /* or whatever you want to call it */
    struct mcp2210_board_config board_config;
    struct mcp2210_chip_settings chip_settings;
    struct mcp2210_chip_settings powerup_chip_settings;
    struct mcp2210_spi_xfer_settings spi_settings;
    struct mcp2210_spi_xfer_settings powerup_spi_settings;
    struct mcp2210_usb_key_params usb_key_params;
};

static int mcp2210_parse_dts(struct mcp2210_device *dev)
{
    struct mcp2210_board_config *board_config;
    /* etc. */
    struct mcp2210_dts_data d* = kzalloc(sizeof(*d), GFP_KERNEL);
    if (!d) /* handle failure */
    board_config = &d->board_config;
    /* etc. */
```

This should also result in smaller code as well since we don't call memset individually for each struct.